### ISSUE TITLE
NewMessage event stop getting update

### ISSUE THREAD

**Issue by batara666** [gram-js/gramjs#280]
I have 1 account logged in GramJS and Telegram Desktop

on the first start, GramJS was working fine, receiving new update, etc, but after a while and I do some action on Telegram Desktop, NewMessage  event stop giving me update, like, there is no update come in...

but, after I use the same TelegramClient for sending message, the NewMessage are working again

my code really as simple as:

```js
async function eventPrint(event) {
  console.log(event)
}
client.addEventHandler(eventPrint, new NewMessage({}));
```

**Comment by painor** [comment]
You need to call a high level request so that telegram knows you want to receive updates. 
SendMessage is one of those. 
Another one is client.getMe(). you can call that after you .connect each time

**Comment by batara666** [comment]
but, It is safe to do?

```js
setInterval(client.getMe, 5000)
```

**Comment by batara666** [comment]
```js
async function eventPrint(event) {
  await client.getMe();
  ....
}
```

weird, still not working

**Comment by painor** [comment]
as I said in the previous comment it should be after connecting.
client = new TelegramClient(.....
await client.connect()
await client.getMe()

**Comment by batara666** [comment]
yes, I do that in the first place, but still the event not receiving any update (like the first case)

**Comment by painor** [comment]
could you share full code?

**Comment by batara666** [comment]
Sorry for long delays

```js
  await client.start({
    phoneNumber: async () => await input.text("Please enter your number: "),
    password: async () => await input.text("Please enter your password: "),
    phoneCode: async () =>
      await input.text("Please enter the code you received: "),
    onError: (err) => console.log(err),
  });
  await client.getMe();
  client.addEventHandler(eventPrint, new NewMessage({}));
```

**Comment by painor** [comment]
are you using an old version of the library maybe? 
does anything suspicious show in the logs?

**Comment by batara666** [comment]
package.json file:

```js
{
  "name": "telegram-mirror",
  "version": "1.0.0",
  "description": "",
  "main": "tg-forwarder.js",
  "scripts": {
    "start": "nodemon tg-forwarder.js"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "express": "^4.17.3",
    "mime-types": "^2.1.34",
    "minio": "^7.0.26",
    "mongodb": "^4.3.1",
    "readline-promise": "^1.0.5",
    "tdl": "^7.0.0",
    "tdl-tdlib-addon": "^1.0.1",
    "uuidv4": "^6.2.12"
  },
  "devDependencies": {
    "input": "^1.0.1",
    "telegram": "^2.5.22"
  }
}
```

**Comment by painor** [comment]
can you share the logs when that happens?

**Comment by batara666** [comment]
[out.log](https://github.com/gram-js/gramjs/files/8319436/out.log)

like, nothing strange there

**Comment by batara666** [comment]
oops, my settings was:

```
  connectionRetries: 5,
  useWSS: true,
```

**Comment by painor** [comment]
Are you running the same session twice ?

**Comment by batara666** [comment]
no sir, only one

**Comment by painor** [comment]
then I honestly have no idea, 
Maybe you should try running client.getMe() every minute to wake up the script.

**Comment by pheianox** [comment]
I think maybe its something with the Telegram API, because I'm getting the same issue

No updates even after doing all the recommendations

I can send messages, can do anything I want, but sadly no updates are coming

**Comment by pheianox** [comment]
As I said it before, the New Message updates are restricted, they do come from private chats, channel posts, but not from channel discussions (comment section), which is kinda sad

**Comment by devagarwal007** [comment]
I am not getting NewMessage event triggered of new channels, seems not related to timeout, I am receiving the new message trigger for old channels but not for the new channels. can someone please help here? It seems the session is alive but there is some internal issue.

**Comment by awdr74100** [comment]
Hi everyone, I'm currently working on a tool to forward group messages, but the development is stalled due to this issue, is there a better way to resolve this?

I can't listen to any messages after 2 minutes, but everything was fine before that.

I'm not good at python, can anyone here confirm that this problem also occurs with telethon?

**Comment by SanSaich** [comment]
I tried everything I could, nothing worked. I switched to the TDL library, there is no such problem there.

**Comment by gee666** [comment]
I have a similar issue: some updates come in, some don't, it seems to be random... I try to receive channel posts.

**Comment by hydra1197** [comment]
Same issue

**Comment by beautyfree** [comment]
Same issue, has anyone found a solution? 
Maybe desktop clients are sending some kind of ping?
